src/lib/operators/table_scan/column_between_table_scan_impl.cpp

```diff
@@ -86,7 +103,10 @@ void ColumnBetweenTableScanImpl::_scan_dictionary_segment(const BaseDictionarySe
     return;
   }
 
-  if (left_value_id >= static_cast<ValueID>(segment.unique_values_count()) || left_value_id == right_value_id) {
+  if (left_value_id == INVALID_VALUE_ID || left_value_id >= static_cast<ValueID>(segment.unique_values_count()) ||
+      left_value_id >= right_value_id) {
+    // TODO(all)
+    // if (left_value_id >= static_cast<ValueID>(segment.unique_values_count()) || left_value_id == right_value_id) {
```

Comment:
?

Reply:
I did not know how to merge this statement with the if-clause above, because I do not fully understand, what is happening here